src/main/java/com/hms_networks/americas/sc/extensions/system/time/LocalTimeOffsetCalculator.java

```diff
@@ -37,44 +42,70 @@ public static synchronized long getLocalTimeOffsetMilliseconds() {
    * #getLocalTimeOffsetMilliseconds()} method.
    *
    * @return calculated local time offset (in milliseconds)
-   * @throws IOException if unable to perform script expression export block descriptor call
+   * @throws InterruptedException if interrupted while attempting script expression export block
+   *     descriptor call
    * @throws ParseException if unable to parse script expression export block descriptor call result
    *     (local time)
    */
   public static synchronized long calculateLocalTimeOffsetMilliseconds()
-      throws IOException, ParseException {
+      throws InterruptedException, ParseException {
     // Export the local time result file
-    String localTime = exportLocalTime();
+    exportLocalTime();
 
     // Parse local time for offset and store/return
-    timeOffsetMilliseconds = parseLocalTimeResult(localTime);
+    timeOffsetMilliseconds = parseLocalTimeResult();
     return timeOffsetMilliseconds;
   }
 
   /**
    * Gets and returns the local time using the script expression export block descriptor call
    * specified by {@link TimeLibConstants#TIME_OFFSET_LOCAL_TIME_EBD}.
    *
-   * @throws IOException if unable to perform script expression export block descriptor call
+   * @throws InterruptedException if interrupted while attempting script expression export block
+   *     descriptor call
    */
-  private static synchronized String exportLocalTime() throws IOException {
+  private static synchronized void exportLocalTime() throws InterruptedException {
     // Build full export block descriptor call and file path
     final String exportBlockDescriptorCall = TimeLibConstants.TIME_OFFSET_LOCAL_TIME_EBD;
 
-    // Run export block descriptor call to get local time
-    Exporter exporter = new Exporter(exportBlockDescriptorCall);
-    return StringUtils.getStringFromInputStream(exporter, "UTF-8");
+    // Run export block descriptor call and retry on failure
+    AutomaticRetryCode automaticRetryCode =
+        new AutomaticRetryCodeLinear() {
+          protected long getLinearSlopeMillis() {
+            return SCTimeUnit.SECONDS.toMillis(2);
+          }
+
+          protected long getMaxDelayMillisBeforeRetry() {
+            return SCTimeUnit.SECONDS.toMillis(30);
+          }
+
+          protected int getMaxRetries() {
+            return 20;
+          }
+
+          protected void codeToRetry() {
+            try {
+              Exporter exporter = new Exporter(exportBlockDescriptorCall);
```

Comment:
Have you tested exporter.finalize() ? This method is listed in the API docs, but has no description. I guessed it was some sort of blocking call to ensure the EBD export had finished.

Reply:
I have not tested the finalize() method, but I'm also not sure what it does. There is a ```.finalize()``` method in the `Object` class in Java, so it could be that.

Reply:
![image](https://user-images.githubusercontent.com/51409174/149527201-3a347842-01a6-436a-9894-b4cb36244005.png)
We don't have access to it anyways